src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figure implements HasArea, Drawable {
+    //limit of figure size when generating randomly
```

Comment:
read the task description carefully: all figures have state - color. So, you don't need getRandomDimension() logic here